docker/Dockerfile-build

```diff
@@ -36,8 +36,8 @@ RUN sudo apt-get update -qq \
 
 RUN rustup update \
     # This specific version of Rust is required for compatibility with mozilla-central
-    && rustup install 1.43.0 \
-    && rustup default 1.43.0
+    && rustup install 1.50.0 \
+    && rustup default 1.50.0
```

Comment:
Rust `1.43` doesn't even have `&&` in const expressions, but we're moving to a min of `1.50` in appservices for unrelated reasons, so this bump seems fine to me.